### ISSUE TITLE
Phase B — DreamerV3 RSSM 完整迁移（StochasticHead + KL balancing + FiLM 注入）

### ISSUE THREAD

**Issue by sqc2290318555** [multica-ai/multica#7610]
# Phase B — DreamerV3 RSSM 完整迁移（代码改动）

> 父 issue: #7608 — CR-DiLA FDM 架构优化：DreamerV3 RSSM 迁移

## 目标

在 CR-DiLA 确定性 FDM 基础上，新增 DreamerV3 式随机路径（Categorical latent + KL balancing），突破 32-dim 瓶颈和 MSE modal averaging 天花板。

### 当前 CR-DiLA（确定性单路径）

```
C-RADIO (1152d, frozen) → CANNEncoder (1152→32) → g [B,T,60,32,32]（确定性, per-token）
→ FDM: g_prev + action → STTransformer → delta_g → g_gen = g_prev + delta_g
→ FusionDecoder: g_gen + content_mem(Mamba 256d) → embedding_gen (1152d)
→ RAE (frozen) → image
```

### 目标 RSSM（确定性 + 随机双路径）

```
确定性路径（保留现有 FDM）:
  h_t = g_prev + FDM(g_prev, a_t)   ← per-token 32-dim, 携带空间细节

随机路径（新增）:
  posterior: q(z_t | h_t, o_t) = Categorical(32 classes × 4 groups)  ← 训练时用
  prior:     p(z_t | h_t, a_t) = Categorical(32 classes × 4 groups)  ← 推理时用
  z_t ~ q (训练) / p (推理), 128 bits 全局

Decoder:
  embedding_gen = FusionDecoder(h_t, content_mem, z_t)  ← z_t 通过 FiLM 注入
```

### 1. `model.py` — 新增 StochasticHead

新增两个子模块：

**PosteriorEncoder**（观测条件后验）
- 输入：当前帧 C-RADIO embedding `o_t`（1152d, 已由 structure_encoder 编码）
- 输出：Categorical logits → `q(z_t | o_t)`，shape `[B, 4, 32]`（4 groups × 32 classes）
- 实现：MLP → reshape → Categorical

**PriorPredictor**（纯转移先验）
- 输入：确定性结构 `h_t`（32d per-token）+ `a_t`（action_dim）
- 输出：Categorical logits → `p(z_t | h_t, a_t)`，shape `[B, 4, 32]`
- 实现：全局 pooling（mean over tokens）→ MLP → Categorical

**RSSM forward 逻辑**（新增方法 `rssm_forward`）
- 训练：z ~ posterior，同时算 prior，KL loss = 0.8*KL(q‖p_prior) + 0.2*KL(q‖p_post)
- 推理：z ~ prior，无 posterior（无未来泄露）

### 2. `forward_model.py` — 新增 PriorHead

在 `AdaForwardDynamics` 中新增 `prior_head`：
- 输入：FDM 输出的 hidden representation（hidden_dim=512，在 to_delta 之前）
- 输出：Categorical logits `[B, 4, 32]`
- 与 FDM 共享 transformer backbone，仅新增输出头（zero-init）

### 3. `fusion_decoder.py` — z_stochastic FiLM 注入

在 `FusionDecoder` 的 `DualAttentionTransformer` 中：
- 新增 `z_proj`: Linear(128, dim_context * 2) → scale, shift
- z_stochastic（128d, one-hot 或 Gumbel-softmax 采样）→ FiLM 调制每层
- zero-init → 训练初始等价于无随机路径（兼容 Phase 1 权重）

### 4. `dila_setup.py` — KL balancing loss

新增 loss 项：
- `kl_loss = 0.8 * KL(q || stop_grad(prior)) + 0.2 * KL(stop_grad(q) || prior)`
- 权重 `kl_weight`（config 新增，默认 0.1）
- 加入总 loss: `total_loss += kl_weight * kl_loss`

### 5. config 新增参数

| 参数 | 默认值 | 说明 |
|---|---|---|
| `use_rssm` | True | 是否启用随机路径 |
| `stoch_classes` | 32 | Categorical 类别数 |
| `stoch_groups` | 4 | Categorical 组数（总 bits = 32×4 = 128） |
| `kl_weight` | 0.1 | KL balancing loss 权重 |
| `kl_balancing_alpha` | 0.8 | prior 权重（1-alpha = posterior 权重） |

## 关键设计决策

1. **全局随机 latent 而非 per-token**：1920 个 token 各加 32×32 Categorical 计算量爆炸。全局 128 bits 足够表达多模态选择（左转/右转/直行），空间细节由确定性 32-dim per-token 承担。

2. **Posterior 仅训练时用**：训练 `z ~ q(o_t)` 收敛快，推理 `z ~ p(h_t)` 无未来泄露。KL 拉近 prior → posterior。

3. **Zero-init 兼容 Phase 1**：所有新分支初始权重 zero-init，训练初始等价于原确定性路径，不破坏 Phase 1 收敛权重。

4. **Gumbel-softmax 采样**：Categorical 采样用 Straight-Through Gumbel-Softmax（前向 one-hot，反向梯度可流通）。

## 执行步骤

1. preflight 确认代码位置
2. checkout `crdila-clean` 分支，创建新分支 `crdila-dreamerv3`
3. 实现 5 个改动（4 个文件 + config）
4. 从 Phase 1 ckpt 续训，前 1000 步观察 KL loss 是否收敛
5. rollout PSNR 目标 > 28 dB

## 预期结果

- **目标**：28-30 dB（接近 RAE ceiling 31.3 dB，E2E gap 缩小到 1-3 dB）
- **风险**：KL loss 不收敛或 posterior collapse → 调整 kl_weight 和 kl_balancing_alpha
- **回退**：zero-init 设计保证可随时通过 `use_rssm=False` 回退到确定性路径

## 代码位置参考

- 模型: `COMPASS/src/compass/models/modules/dila/cr_dila/model.py`（544 行）
- FDM: `COMPASS/src/compass/models/modules/dila/cr_dila/forward_model.py`
- FusionDecoder: `COMPASS/src/compass/models/modules/dila/cr_dila/fusion_decoder.py`
- 训练 loss: `COMPASS/src/compass/models/modules/dila/dila_setup.py`（L1540-1960）
- 配置: `COMPASS/projects/vla/configs/cluster/dila_crdila_phase2_fulldata_5090_1x8.py`

**Comment by sqc2290318555** [comment]
已迁移到 Multica tracked issue（GH #602 (Phase B)). 原 issue 通过 gh CLI 创建，未被 Multica 跟踪。已在 Multica 重新创建并分配给 prime-mda-orchestrator agent。
